diff -urN gqview-2.1.3/configure.in gqview-2.1.3.new/configure.in
--- gqview-2.1.3/configure.in	2006-11-04 23:15:52.000000000 +0100
+++ gqview-2.1.3.new/configure.in	2006-11-04 23:20:35.000000000 +0100
@@ -25,7 +25,7 @@
 dnl checks for functions
 AC_CHECK_FUNCS(strverscmp)
 
-ALL_LINGUAS="ar be bg ca cs da de eo es et eu fi fr hu id it ja ko nl no pl pt_BR ro ru sk sl sv th tr uk vi zh_CN.GB2312 zh_TW"
+ALL_LINGUAS="ar be bg ca cs da de eo es et eu fi fr hu id it ja ko nl nb pl pt_BR ro ru sk sl sv th tr uk vi zh_CN zh_TW"
 GETTEXT_PACKAGE=$PACKAGE
 AC_SUBST(GETTEXT_PACKAGE)
 AM_GLIB_GNU_GETTEXT
