diff -burN gqview-1.5.2.orig/gqview.desktop gqview-1.5.2/gqview.desktop
--- gqview-1.5.2.orig/gqview.desktop	2004-08-28 11:14:25.844191184 +0200
+++ gqview-1.5.2/gqview.desktop	2004-08-28 11:19:24.938721904 +0200
@@ -5,8 +5,10 @@
 GenericName[tr]=Resim göstericisi
 GenericName[fr]=Explorateur d'images
 GenericName[hu]=Képnéző
+GenericName[pl]=Przeglądarka obrazków
 Comment=View and manage images
 Comment[hu]=Képek megjelenítése és rendszerezése
+Comment[pl]=Przeglądanie i manipulowanie obrazkami
 Exec=gqview
 Icon=gqview.png
 Type=Application
@@ -14,4 +16,5 @@
 StartupNotify=false
 StartupWMClass=gqview
 Encoding=UTF-8
-Categories=Application;Graphics;
+Categories=GTK;Graphics;Viewer;
+# vi: encoding=utf-8
