diff -urN gqview-1.5.4.orig/configure.in gqview-1.5.4/configure.in
--- gqview-1.5.4.orig/configure.in	2004-11-11 10:27:08.036267024 +0100
+++ gqview-1.5.4/configure.in	2004-11-11 10:31:50.413339184 +0100
@@ -56,7 +56,7 @@
 localedir='${datadir}/locale'
 AC_SUBST(localedir)
 
-ALL_LINGUAS="ar bg cs da de es et fi fr hu id it ja nl no pl pt_BR ro ru sk sl sv th tr uk vi zh_CN.GB2312 zh_TW"
+ALL_LINGUAS="ar bg cs da de es et fi fr hu id it ja nb nl pl pt_BR ro ru sk sl sv th tr uk vi zh_CN zh_TW"
 AM_GNU_GETTEXT
 
 AC_SUBST(CFLAGS)
