diff -ruN gqview-1.5.7./configure.in gqview-1.5.7/configure.in
--- gqview-1.5.7./configure.in	2005-02-06 22:51:59.000000000 +0100
+++ gqview-1.5.7/configure.in	2005-02-06 22:56:17.665046896 +0100
@@ -56,7 +56,7 @@
 localedir='${datadir}/locale'
 AC_SUBST(localedir)
 
-ALL_LINGUAS="ar be bg ca cs da de es et fi fr hu id it ja ko nl no pl pt_BR ro ru sk sl sv th tr uk vi zh_CN.GB2312 zh_TW"
+ALL_LINGUAS="ar be bg ca cs da de es et fi fr hu id it ja ko nl nb pl pt_BR ro ru sk sl sv th tr uk vi zh_CN zh_TW"
 GETTEXT_PACKAGE=$PACKAGE
 AC_SUBST(GETTEXT_PACKAGE)
 AM_GLIB_GNU_GETTEXT
