--- ./src/filelist.c.org	2006-11-07 21:15:22.000000000 +0100
+++ ./src/filelist.c	2008-02-24 20:56:58.632165946 +0100
@@ -217,6 +217,7 @@ void filter_add_defaults(void)
 	filter_add_if_missing("nef", "Nikon raw format", ".nef", TRUE);
 	filter_add_if_missing("orf", "Olympus raw format", ".orf", TRUE);
 	filter_add_if_missing("pef", "Pentax raw format", ".pef", TRUE);
+	filter_add_if_missing("dng", "Adobe Digital Negative", ".dng", TRUE);
 }
 
 static GList *filter_to_list(const gchar *extensions)
